src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -467,8 +499,21 @@ else if (type.isType(ConsumerTypeEnum.HYPERVISOR)) {
         return entitlementVersion != null && entitlementVersion.startsWith("3.");
     }
 
-    private String createDN(Consumer consumer, Owner owner) {
-        return "CN=" + consumer.getUuid() + ", O=" + Rdn.escapeValue(owner.getKey());
+    private String createDN(String commonName, String organizationName) {
+        StringBuilder builder = new StringBuilder();
+        if (commonName != null && !commonName.isEmpty()) {
+            builder.append("CN=" + commonName);
```

Comment:
So I dug into this a bit, and it looks like the concatenation is indeed being done before the append call. Using the test class:

```
public class TestClass {

    public static void main(String[] args) {
        StringBuilder builder = new StringBuilder("test - ");

        if (args.length > 1) {
            builder.append("arg count: " + args.length);
        }
        else {
            builder.append("no args :(");
        }

        System.out.println(builder.toString());
    }

}
```

When we compile that down and then look at the resultant bytecode, the important bits on line 7 look as follows:

```
      18: arraylength
      19: invokedynamic #14,  0             // InvokeDynamic #0:makeConcatWithConstants:(I)Ljava/lang/String;
      24: invokevirtual #18                 // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
```

The compiler is using some [very magical dynamically created function](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/invoke/StringConcatFactory.html#makeConcatWithConstants(java.lang.invoke.MethodHandles.Lookup,java.lang.String,java.lang.invoke.MethodType,java.lang.String,java.lang.Object...)) to generate an optimized string concatenation function for handling that specific concatenation.

When we turn this into a chain of appends, however, we get this:
```
      17: ldc           #14                 // String arg count:
      19: invokevirtual #16                 // Method java/lang/StringBuilder.append:(Ljava/lang/String;)Ljava/lang/StringBuilder;
      22: aload_0
      23: arraylength
      24: invokevirtual #20                 // Method java/lang/StringBuilder.append:(I)Ljava/lang/StringBuilder;
```

So, it looks like the compiler *won't* reuse an existing/active StringBuilder to handle concatenation within the arguments going into the string builder.

As to whether or not that's faster or slower... I don't know. Last I looked into this, the compiler was straight up replacing concatenation operations with inline StringBuilders itself, so the comparison was a bit more clear cut. I have no idea what this new generated concat factory method is doing under the hood, but based on the fact that it's still string concatenation at the end of the day, I have to assume that it's not going to be any faster than just appending to our target string builder. Another test for another day, I suppose.

Reply:
It used to be compiled to StringBuilder append chains in Java 8. Since Java 9 compiler handles concatenation with invokeDynamic call to [StringConcatFactory](https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/StringConcatFactory.html).

Reply:
This was interesting! Updated